test/uk/gov/hmrc/hec/services/FileCreationServiceSpec.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.hec.services
+
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import play.api.test.Helpers._
+import uk.gov.hmrc.hec.models.licence.{LicenceTimeTrading, LicenceType, LicenceValidityPeriod}
+import uk.gov.hmrc.hec.util.TimeProvider
+
+import java.time.{LocalDate, LocalTime, ZoneId}
+
+class FileCreationServiceSpec extends AnyWordSpec with Matchers with MockFactory {
+  val mockTimeProvider = mock[TimeProvider]
+
+  val fileCreationService = new FileCreationServiceImpl(mockTimeProvider)
+
+  def mockDateProviderToday(d: LocalDate) = (mockTimeProvider.currentDate _).expects().returning(d)
+  def mockTimeProviderToday(d: LocalTime) = (mockTimeProvider.currentTime(_: ZoneId)).expects(*).returning(d)
```

Comment:
we could actually say what `ZoneId` we're expecting if we know it. Also, the def here and above don't both return today - maybe they should be called something like `mockTimeProviderToday` and `mockTimeProviderNow or `mockTimeProviderCurrentTime`?

Reply:
that's typo 🙈 i was planning to call it mockTimeProviderNow. Will fix it